parts/k8s/addons/kubernetesmasteraddons-calico-daemonset.yaml

```diff
@@ -137,8 +137,7 @@ data:
           "nodename": "__KUBERNETES_NODE_NAME__",
           "mtu": 1500,
           "ipam": {
-              "type": "host-local",
```

Comment:
This is the other area that needs some design input - right now I've adjusted the config to be hardcoded to use the azure-vnet-ipam plugin. However, this means the old "Calico + kubenet" approach won't work anymore.

I think we need a way to modify this config based on the networking type specified, but not sure what the best way to do that is. Any thoughts?

Reply:
Hi @caseydavenport. To make certain parts of the calico YAML configurable, we add placeholders in the YAML file. For the user to provide their desired config, they specify it as part of the Add-on Config in their apimodel. You can check [this](https://github.com/Azure/acs-engine/blob/1f1dda675dc168ee46a47ab3a572f8871496272c/pkg/api/types.go#L90). After, you will find that the subsitution logic for these placeholders are found [here](https://github.com/Azure/acs-engine/blob/1f1dda675dc168ee46a47ab3a572f8871496272c/parts/k8s/kubernetesmastercustomdata.yml#L361).

Reply:
Also check L345 of this file, you will see that we use the Calico IPv4 as the placeholder here